internal/db/db.go

```diff
@@ -0,0 +1,55 @@
+package db
+
+import (
+	"github.com/smailic05/TaskInfoblox/internal/model"
+	"gorm.io/gorm"
+)
+
+type DB struct {
+	*gorm.DB
+}
+
+func NewDB(db *gorm.DB) *DB {
+	return &DB{db}
+}
+
+func (db *DB) Load(user model.User) ([]model.User, error) {
+	users := make([]model.User, 0)
+	result := db.Where("username LIKE ? AND address LIKE ? AND phone LIKE ?", user.Username, user.Address, user.Phone).Find(&users)
```

Comment:
Here and below: some things can be simplified and shorted. For example, instead or `result`, you can use `err`:
```
err := db.Where("username LIKE ? AND address LIKE ? AND phone LIKE ?", user.Username, user.Address, user.Phone).Find(&users).Error
```
and then, just return the results:
```
return users, err
```
But if you're still going to check for the errors, then you probably may return nil value instead of the empty slice. The same approach can be used in some cases below

Reply:
Fixed that